### PR TITLE
Give same rank to users with same scores in SAP leaderboard

### PR DESCRIPTION
**Pull request by erilyth**

### CODE REVIEW

File: src/app/views/sap/leaderboard.php [hunk 1]
```diff
@@ -17,16 +17,20 @@
             <th>Score</th>
         </tr>
         <?php
-            $rank = 1;
+            $rank = 0;
+            $score_prev = $user_scores[0]['score']+1;
             foreach ($user_scores as $user_score) {
                 $name = $user_score['name'];
                 $score = $user_score['score'];
+                if ($score_prev>$score){
```
Comment (polybuildr) [inline src/app/views/sap/leaderboard.php]: Not a very big issue, but please put in a space before and after the `>`.

**Comment by nisargjhaveri** [comment]
Why does this have two commits?

**Comment by polybuildr** [comment]
I'm guessing it's a rebasing issue.

Could you try pulling in master, then making a new branch, and then cherry picking this commit? Unless @nisargjhaveri has a better idea?

**Comment by polybuildr** [comment]
Oh. Just squashing them _might_ work too. :P

**Comment by erilyth** [comment]
Squashed them :P

**Comment by polybuildr** [comment]
Asked him to make a few changes. Reviewing now.

File: src/app/views/sap/leaderboard.php [hunk 2]
```diff
@@ -18,15 +18,19 @@
         </tr>
         <?php
             $rank = 1;
+            $score_prev = $user_scores[0]['score'];
```
Comment (polybuildr) [inline src/app/views/sap/leaderboard.php:21]: `Fatal error: Cannot use object of type mysqli_result as array`.

**Comment by polybuildr** [comment]
Yes, now it works. :P

File: src/app/views/sap/leaderboard.php (hunk 2 above)
Comment (polybuildr) [inline src/app/views/sap/leaderboard.php:21]: Handled in updated PR.

**Comment by polybuildr** [comment]
Filed http://phab.felicity.iiit.ac.in/T134 sort of related to this. This in itself looks fine.

**Pull request merged**